### DISCUSSION TITLE
Shadow appears in AR even though shadow-intensity=0

### DISCUSSION THREAD

**Discussion by alecjacobson** [google/model-viewer#3928] [Q&A]
### Description

Shadow appears in AR even thought shadow-intensity=0

#### Live Demo

Go to https://modelviewer.dev/editor/ , select the Astronaut (for example), set Shadow Intensity slider to 0. Then open in mobile. In the scene viewer it looks correct (no shadow), but when entering AR mode the shadow is back.

I'm using a Google Pixel 3 in Chrome.

### Version

- model-viewer: v2.0.2 (I'm guessing the editor is the most recent release)

### Browser Affected

- [x] Chrome, version: xx.x.xxxx.xx
- [ ] Edge
- [ ] Firefox
- [ ] IE
- [ ] Safari

### OS

- [x] Android
- [ ] iOS
- [ ] Linux
- [ ] MacOS
- [ ] Windows

### AR

- [x] WebXR
- [ ] SceneViewer
- [ ] QuickLook

**Comment by elalish** [comment]
This is by design; we always show a shadow in AR since the model is always placed on the floor. Is there a use case where you'd need to disable that?

**Reply by alecjacobson** [reply]
I can think of at least three reasons why this is an issue:

1) when using baked shadows. The model-viewer shadows are OK, but often it's possible to bake out a better shadow. I believe this was also one of the motivations for these options in the first place.

2) the current model-viewer shadows seem to use a "ground-plane" computed once based on the rest geometry. This mean it will be wrong for animations that move below that plane. This is a separate issue, but a hotfix is to just turn off the shadows. Currently that setting is not being respected in the AR mode because the AR mode overrides the explicit shadow parameters.

3) When the lighting isn't from above. The default shadow is reasonablish for most settings, but even if a model is on the floor it doesn't mean the lighting is from directly above and this can look more wrong than having no shadow.

**Reply by elalish** [reply]
Thanks, those are good points. Of course we can only affect WebXR mode, so iOS would be even more inconsistent. PRs welcome if you have some ideas, though we'll have to be careful of breaking changes.

**Reply by matt-deboer** [reply]
Another potential reason, setting shadow-intensity very low is one hack to correct an issue with transmissive/transparent objects.

The upper image has 'shadow-intensity="0.1"', which is ignored in AR mode (bottom), showing an intense dark shadow which doesn't match the amount of light blocked by the transparent tray (viewed on a Pixel 4a).

![image](https://github.com/google/model-viewer/assets/1491381/a461d024-3283-408a-82c5-a905c4b11daa)
![image](https://github.com/google/model-viewer/assets/1491381/9312eb8e-ef0b-447f-96d8-2858db81b8ef)

**Reply by elalish** [reply]
PRs welcome - better to discuss in the context of something testable.
